sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1334,12 +1334,19 @@ object EliminateOuterJoin extends Rule[LogicalPlan] with PredicateHelper {
     val rightConditions = splitConjunctiveConditions
       .filter(_.references.subsetOf(join.right.outputSet))
 
+    val joinAttributeSet = if (join.condition.isDefined) {
```

Comment:
This might shorter: `join.condition.map(_.references).getOrElse(AttributeSet.empty)`